app/models/katello/concerns/http_proxy_extensions.rb

```diff
@@ -4,7 +4,10 @@ module HttpProxyExtensions
       extend ActiveSupport::Concern
 
       included do
+        has_many :root_repositories, :class_name => "::Katello::RootRepository", :foreign_key => :http_proxy_id,
+          :inverse_of => :http_proxy, :dependent => :nullify
         after_save :update_default_proxy_setting
+        after_save :update_repository_proxy_details
```

Comment:
this should probably be after_update ,and actually update_default_proxy_setting, likely should be too